Netbox_Route53/Netbox_route53.py

```diff
@@ -75,74 +82,74 @@ def __init__(self):
         self.R53_Record = json.dumps(R53_get_response)
         self.R53 = json.loads(self.R53_Record)
 
+        self.r53_record_dict = {}
+        self.r53_tag_dict = {}
+
         self.dns_string = ""
         for i in self.nb_ip_addresses:
             nb_dns = i.dns_name + "." + self.HZ_Name
             self.dns_string += nb_dns
 
     def get_nb_records(self):
-        timespan = datetime.today() - timedelta(hours=100, minutes=0)
+        timespan = datetime.today() - timedelta(days=self.timespan)
         timespan.strftime('%Y-%m-%dT%XZ')
         ip_search = self.nb.ipam.ip_addresses.filter(within=self.nb_ip_addresses, last_updated__gte=timespan)
         return ip_search
 
     def check_record_exists(self, dns, ip):
-        nb_dns = '''"''' + dns + '''"'''
-        r53_dns = '''"Name": ''' + nb_dns
-        nb_ip = '''"''' + ip + '''"'''
-        r53_ip = '''"Value": ''' + nb_ip
-        if r53_dns in self.R53_Record or r53_ip in self.R53_Record:
+        if dns in self.r53_record_dict or ip in self.r53_record_dict.values():
             return True
         return False
 
-    def verify_and_update(self, dns, ip):
-        tag = self.get_r53_record_tag(dns)
-        v = self.R53_Record.count('''"Name"''')
+    def get_r53_record_tag(self, dns):
+        if dns in self.r53_tag_dict:
+            return True
+        return None
+
+    def get_r53_records(self):
+        R53_Record_response = self.client.list_resource_record_sets(HostedZoneId=self.r53_zone_id)
+        R53_Record = json.dumps(R53_Record_response)
+        R53 = json.loads(R53_Record)
+        v = R53_Record.count('''"Name"''')
         for n in range(0, v):
-            R53_ip = self.R53['ResourceRecordSets'][n]['ResourceRecords'][0]['Value']
-            R53_Record_name = self.R53['ResourceRecordSets'][n]['Name']
+            R53_Record_name = R53['ResourceRecordSets'][n]['Name']
```

Comment:
You can change this to something like:
```
for r53_record in R53_Record_response['ResourceRecordSets']:
```
which will iterate over that dictionary, and lets get rid of all the json dump/load stuff, as well as making some of the follow code cleaner